tooling/minimize/src/rvalue.rs

```diff
@@ -172,10 +173,17 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
             smir::Rvalue::Len(place) => {
                 // as slices are unsupported as of now, we only need to care for arrays.
                 let ty = place.ty(&self.locals_smir).unwrap();
-                let Type::Array { elem: _, count } = self.translate_ty_smir(ty, span) else {
-                    panic!()
-                };
-                ValueExpr::Constant(Constant::Int(count), <usize>::get_type())
+                match self.translate_ty_smir(ty, span) {
+                    Type::Array { elem: _, count } =>
+                        ValueExpr::Constant(Constant::Int(count), <usize>::get_type()),
+                    Type::Slice { .. } =>
+                        build::get_metadata(build::addr_of(
+                            self.translate_place_smir(place, span),
+                            build::raw_ptr_ty(PointerMetaKind::ElementCount),
+                        )),
+
+                    _ => panic!("RValue::Len only supported for arrays & slices"),
```

Comment:
```suggestion
                    _ => rs::span_bug!(span, "RValue::Len only supported for arrays & slices"),
```
while we are at it, might as well make this a nicer error